### PR TITLE
Load default XML only once [10188]

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
When the environment variable `FASTRPS_DEFAULT_PROFILES_FILE` is set pointing to the `DEFAULT_FASTRTPS_PROFILES.xml` file in the current directory, the file is loaded twice and LogErrors are shown stating that the profile already exists. This PR corrects the issue by using the absolute path when the default file is loaded. It also includes a regression test.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3437/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/631/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3063/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3471/)

File: test/unittest/xmlparser/XMLLoadFileTests.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <thread>
+#include <mutex>
+#include <fstream>
```
Comment (imontesino) [inline test/unittest/xmlparser/XMLLoadFileTests.cpp:18]:
```suggestion

#include <mutex>
#include <fstream>
```

Unused header

File: src/cpp/xmlparser/XMLParserImpl.hpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef FASTRTPS_XMLPARSER_XMLPARSERIMPL_H_
+#define FASTRTPS_XMLPARSER_XMLPARSERIMPL_H_
+
+#include <fastrtps/xmlparser/XMLParser.h>
```
Comment (imontesino) [inline src/cpp/xmlparser/XMLParserImpl.hpp:18]:
```suggestion

#include <string>

#include <fastrtps/xmlparser/XMLParser.h>
```
Following the guideline "Include what you use".

**Review by imontesino — CHANGES_REQUESTED** [review]
CI is failing, although test is passing on local. @JLBuenoLopez-eProsima  Could you look into that?

**Comment by JLBuenoLopez-eProsima** [comment]
> CI is failing, although the test is passing on local. @JLBuenoLopez-eProsima Could you look into that?

The new test depends on capturing a Log INFO entry. It seems that CI is not running with this kind of entries enabled. Thus the reason why the new test is failing in all platforms.

File: test/unittest/xmlparser/XMLLoadFileTests.cpp (hunk 1 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/xmlparser/XMLLoadFileTests.cpp]: Done in e035ea991e1381ba57187339de12e6a3dc686370

File: src/cpp/xmlparser/XMLParserImpl.hpp (hunk 2 above)
Comment (JLBuenoLopez-eProsima) [reply src/cpp/xmlparser/XMLParserImpl.hpp:20]: Done in e035ea991e1381ba57187339de12e6a3dc686370

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3458/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/651/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3079/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3492/)

**Comment by MiguelCompany** [comment]
@JLBuenoLopez-eProsima Now that #1704 has been merged, we can finish this one

**Comment by MiguelCompany** [comment]
Adding `skip-ci` as CI will be launched manually

**Comment by JLBuenoLopez-eProsima** [comment]
CI results:

* [Linux CI](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3715/):SUCCESS
* [Windows CI](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3737/): SUCCESS
* [Mac CI](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3268/): Warnings already solved in Master in #1525.

File: include/fastrtps/xmlparser/XMLProfileManager.h [hunk 3]
```diff
@@ -230,27 +230,31 @@ class XMLProfileManager
 
 private:
 
-    RTPS_DllAPI static XMLP_ret extractDynamicTypes(
+    static XMLP_ret loadXMLFile(
+            const std::string& filename,
+            const bool& is_default);
```
Comment (MiguelCompany) [inline include/fastrtps/xmlparser/XMLProfileManager.h:235]: Why are you using `const bool&` ?

```suggestion
            bool is_default);
```

File: src/cpp/xmlparser/XMLParser.cpp [hunk 4]
```diff
@@ -3104,6 +3244,31 @@ XMLP_ret XMLParser::fillDataNode(tinyxml2::XMLElement* p_profile, DataNode<Subsc
     return XMLP_ret::XML_OK;
 }
 
+XMLP_ret XMLParserImpl::loadXML(
+        const std::string& filename,
+        up_base_node_t& root,
+        const bool& is_default)
```
Comment (MiguelCompany) [inline src/cpp/xmlparser/XMLParser.cpp:3250]:
```suggestion
        bool is_default)
```

File: src/cpp/xmlparser/XMLParserImpl.hpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+     * Load a XML file.
+     * @param filename Name for the file to be loaded.
+     * @param root Root node.
+     * @param is_default Is the default XML file.
+     * @return XMLP_ret::XML_OK on success, XMLP_ret::XML_ERROR in other case.
+     */
+    static XMLP_ret loadXML(
+            const std::string& filename,
+            up_base_node_t& root,
+            const bool& is_default);
```
Comment (MiguelCompany) [inline src/cpp/xmlparser/XMLParserImpl.hpp:44]:
```suggestion
            bool is_default);
```

File: src/cpp/xmlparser/XMLProfileManager.cpp [hunk 6]
```diff
@@ -226,6 +277,13 @@ XMLP_ret XMLProfileManager::loadXMLNode(
 
 XMLP_ret XMLProfileManager::loadXMLFile(
         const std::string& filename)
+{
+    return loadXMLFile(filename, false);
+}
+
+XMLP_ret XMLProfileManager::loadXMLFile(
+        const std::string& filename,
+        const bool& is_default)
```
Comment (MiguelCompany) [inline src/cpp/xmlparser/XMLProfileManager.cpp:286]:
```suggestion
        bool is_default)
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: include/fastrtps/xmlparser/XMLProfileManager.h (hunk 3 above)
Comment (JLBuenoLopez-eProsima) [reply include/fastrtps/xmlparser/XMLProfileManager.h]: You are right. My mistake.

**Review by MiguelCompany — APPROVED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3732/)
* Linux aarch64Build Status
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3285/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3757/)

**Pull request merged**